### PR TITLE
[295] Implement diversity disclosure section

### PR DESCRIPTION
**Pull request by darokel**
### Context

- https://trello.com/c/ljSVTV3Z/295-s-record-if-the-trainee-has-shared-diversity-details-or-not

### Changes proposed in this pull request

- Create the start of the diversity section (disclosure)

Screenshot 2020-10-21 at 08 47 41

Screenshot 2020-10-21 at 08 47 50

### Guidance to review

- Visit https://dfe-twd-rttd-pr-104.herokuapp.com/
- Create a trainee and head to https://dfe-twd-rttd-pr-104.herokuapp.com/trainees/2/diversity/information-disclosed/edit
- Try to continue without making a selection, validations should trigger
- Make a selection, continue and your selection should be saved

### CODE REVIEW

**Comment by darokel** [comment]
[[S] Record if the trainee has shared diversity details or not](https://trello.com/c/ljSVTV3Z/295-s-record-if-the-trainee-has-shared-diversity-details-or-not)

File: spec/services/diversities/disclosures/update_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        let(:trainee) { create(:trainee, diversity_disclosure: nil) }
+        let(:service) { described_class.new(trainee: trainee, attributes: attributes) }
+
+        context "when disclosure attribute is valid" do
+          let(:attributes) { { diversity_disclosure: "yes" } }
+
+          before do
+            service.call
+            trainee.reload
+          end
```
Comment (defong) [inline spec/services/diversities/disclosures/update_spec.rb:16]: can probably move it to the top

```ruby
describe Update do
  describe ".call" do
    let(:trainee) { create(:trainee, diversity_disclosure: nil) }
    let(:service) { described_class.new(trainee: trainee, attributes: attributes) }
    before do
        service.call
        trainee.reload
    end

```

File: spec/services/diversities/disclosures/update_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        end
+
+        context "when disclosure attribute is invalid" do
+          let(:attributes) { { diversity_disclosure: nil } }
+
+          before do
+            service.call
+            trainee.reload
+          end
```
Comment (defong) [inline spec/services/diversities/disclosures/update_spec.rb:34]: can probably move it to the top

```ruby
describe Update do
  describe ".call" do
    let(:trainee) { create(:trainee, diversity_disclosure: nil) }
    let(:service) { described_class.new(trainee: trainee, attributes: attributes) }
    before do
        service.call
        trainee.reload
    end

```

**Review by defong — APPROVED** [review]

File: spec/services/diversities/disclosures/update_spec.rb (hunk 1 above)
Comment (darokel) [reply spec/services/diversities/disclosures/update_spec.rb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/104/commits/8150d6251de7a79e8a063c63b2656f5babac358a

File: spec/services/diversities/disclosures/update_spec.rb (hunk 2 above)
Comment (darokel) [reply spec/services/diversities/disclosures/update_spec.rb]: done https://github.com/DFE-Digital/register-trainee-teachers/commit/8150d6251de7a79e8a063c63b2656f5babac358a

File: app/views/trainees/diversity/disclosures/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds-from-desktop">
+    <%= form_with(model: @disclosure, url: trainee_diversity_disclosure_path(@trainee), method: :put, local: true) do |f| %>
+      <%= f.govuk_error_summary %>
+      
+      <h1 class="govuk-heading-l">Has the trainee shared diversity information?</h1>
+
+      <%= f.govuk_radio_buttons_fieldset(
+        :diversity_disclosure, 
+        legend: { text: "" },
```
Comment (aliuk2012) [inline app/views/trainees/diversity/disclosures/edit.html.erb:17]: The legend should be the page heading. You should be able use `tag: "h1"` option. Give me shout if you need some help

**Review by aliuk2012 — CHANGES_REQUESTED** [review]
Minor issue, but otherwise looks good to go

File: app/views/trainees/diversity/disclosures/edit.html.erb (hunk 3 above)
Comment (darokel) [reply app/views/trainees/diversity/disclosures/edit.html.erb]: done https://github.com/DFE-Digital/register-trainee-teachers/pull/104/commits/01ce1d213a004761f86dc1961b1676983a87e84c

**Review by aliuk2012 — APPROVED** [review]

**Pull request merged**